omg/pastebin.py

```diff
@@ -0,0 +1,56 @@
+from .api import API
+from typing import List, Optional
+from datetime import datetime
+
+
+class Paste:
+    def __init__(self, title: str, content: str, modified_on: str, counter: int, api: API):
+        self._api = api
+        self.title = title
+        self.content = content
+        self.modified_on = datetime.fromtimestamp(modified_on)
+
+    def __str__(self):
+        return f"title: {self.title}, content: {self.content}, modified_on: {self.modified_on.timestamp()}"
+
+    def delete(self):
+        self._api.request(
+            path=f"/address/{self.address}/pastebin/{self.title}",
+            method="DELETE"
+        )
+
+
+class PasteBinRequestor:
+    def __init__(self, api: API, default_username) -> None:
+        self.api: API = api
+        self.default_username = default_username
+
+    def retrieve(self, paste: str, address: Optional[str] = None) -> Paste:
+        un = address or self.default_username
+        if paste is None:
+            raise Exception("please include a paste!")
+        r = self.api.request(f"/address/{un}/pastebin/{paste}")
+        return Paste(**r["response"]["paste"])
+
+    def retrieve_all(self, address=None) -> List[Paste]:
+        un = address or self.default_username
+        r = self.api.request(f"/address/{un}/pastebin")
+        pastes = r['response']['pastes']
+        ls = [Paste(**x) for x in pastes]
+        return ls
+
+    def create(self, title, content, address=None) -> Paste:
```

Comment:
create can also have a parameter called listed (whether the post is public or unlisted), 1 = listed, null = unlisted, could probably be a boolean parameter and then it changes in the actual request making?

Reply:
is it sent in the json as an integer or string?

Reply:
when its 1, its a string, when its null, its just sent as plain null